### PR TITLE
Reduce cxxtools::Json footprint

### PR DESCRIPTION
**Pull request by sinanyavas-eaton**

### CODE REVIEW

File: lib/src/secw_document.cc [hunk 1]
```diff
@@ -406,13 +377,7 @@ void operator<<=(std::string& str, const Document& doc)
...
 
     try {
-        std::stringstream output;
-
-        cxxtools::JsonSerializer serializer(output);
-        serializer.beautify(true);
-        serializer.serialize(si);
-
-        str = output.str();
+        str = JSON::writeToString(si, false);
```
Comment (ericclappier-eaton) [inline lib/src/secw_document.cc]: Please beautify = true in this case

File: lib/src/secw_document.cc [hunk 2]
```diff
@@ -445,13 +407,7 @@ void operator<<=(std::string& str, const std::vector<DocumentPtr>& docs)
...
```
Comment (ericclappier-eaton) [inline lib/src/secw_document.cc]: Please beautify = true in this case

**Review by ericclappier-eaton — DISMISSED** [review]

File: lib/src/cam_helpers.cc [hunk 3]
```diff
@@ -48,11 +44,9 @@ std::string serialize(const cxxtools::SerializationInfo& si)
     std::string returnData("");
 
     try {
-        std::stringstream        output;
-        cxxtools::JsonSerializer serializer(output);
-        serializer.serialize(si);
+        cxxtools::SerializationInfo tmpCpy(si);
+        returnData = JSON::writeToString(tmpCpy, false);
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/cam_helpers.cc]: i think that local cpy is now out of context.
fty-common-json.h api was updated to handle const& si....

File: lib/src/secw_document.cc [hunk 4]
```diff
@@ -406,13 +377,7 @@ void operator<<=(std::string& str, const Document& doc)
...
 
     try {
-        std::stringstream output;
-
-        cxxtools::JsonSerializer serializer(output);
-        serializer.beautify(true);
-        serializer.serialize(si);
-
-        str = output.str();
+        str = JSON::writeToString(si);
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/secw_document.cc]: explicitize beautify as true

File: lib/src/secw_document.cc [hunk 5]
```diff
@@ -445,13 +407,7 @@ void operator<<=(std::string& str, const std::vector<DocumentPtr>& docs)
...
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/secw_document.cc]: explicitize beautify as true

File: lib/src/secw_security_wallet.cc [hunk 6]
```diff
@@ -233,12 +223,7 @@ void SecurityWallet::save() const
     rootSi.addMember("version") <<= SECW_VERSION;
     rootSi.addMember("portfolios") <<= m_portfolios;
 
-    // open the file
-    std::ofstream output(m_pathDatabase.c_str());
-
-    cxxtools::JsonSerializer serializer(output);
-    serializer.beautify(true);
-    serializer.serialize(rootSi);
+    JSON::writeToFile(m_pathDatabase, rootSi);
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/secw_security_wallet.cc]: explicitize beautify as true

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

File: lib/src/cam_exception.cc [hunk 7]
```diff
@@ -20,8 +20,8 @@
 */
 
 #include "cam_exception.h"
-#include <cxxtools/jsondeserializer.h>
-#include <cxxtools/jsonserializer.h>
+#include "cxxtools/serializationinfo.h"
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/cam_exception.cc]: #include <cxxtools/serializationinfo.h>

File: lib/src/cam_exception.cc [hunk 8]
```diff
@@ -89,15 +86,11 @@ const char* CamException::what() const noexcept
...
-    cxxtools::JsonSerializer serializer(output);
-    serializer.beautify(true);
-
     cxxtools::SerializationInfo si;
+
     si <<= *(this);
-    serializer.serialize(si);
 
-    return output.str();
+    return JSON::writeToString(si);
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/cam_exception.cc]: specify beautify = true

File: lib/src/secw_exception.cc [hunk 9]
```diff
@@ -27,8 +27,8 @@
 */
 
 #include "secw_exception.h"
-#include <cxxtools/jsondeserializer.h>
-#include <cxxtools/jsonserializer.h>
+#include "cxxtools/serializationinfo.h"
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/secw_exception.cc]: #include <cxxtools/serializationinfo.h>

File: lib/src/secw_helpers.cc [hunk 10]
```diff
@@ -28,8 +28,8 @@
 
 #include "secw_exception.h"
 #include "secw_openssl_wrapper.h"
-#include <cxxtools/jsondeserializer.h>
-#include <cxxtools/jsonserializer.h>
+#include "cxxtools/serializationinfo.h"
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/secw_helpers.cc]: use <>   instead of ""

File: lib/src/secw_security_wallet.cc [hunk 11]
```diff
@@ -29,8 +29,8 @@
 #include "secw_security_wallet.h"
 #include "fty_security_wallet.h"
 #include "secw_helpers.h"
-#include <cxxtools/jsondeserializer.h>
-#include <cxxtools/jsonserializer.h>
+#include "cxxtools/serializationinfo.h"
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/secw_security_wallet.cc]: <>    (not "")

**Review by FrancoisRegisDegott-eaton — CHANGES_REQUESTED** [review]

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

**Pull request merged**